python/pyarrow/tests/test_pandas.py

```diff
@@ -4523,9 +4550,11 @@ def test_metadata_compat_range_index_pre_0_12():
     gen_name_1 = '__index_level_1__'
 
     # Case 1: named RangeIndex
-    e1 = pd.DataFrame({
-        'a': a_values
-    }, index=pd.RangeIndex(0, 8, step=2, name='qux'))
+    e1 = pd.DataFrame(
+        {'a': a_values},
+        index=pd.RangeIndex(0, 8, step=2, name='qux'),
+        columns=pd.Index(['a'], dtype=object)
```

Comment:
Does the column type created with the dict argument differ from this?

Reply:
This test is specifically using old metadata that specifies the dtype of the columns is object dtype, and then pyarrow tries to restore it that way.

It's the question if we should do that though .. Because every file written from a pandas DataFrame before pandas 3.0 will have that, so maybe we should specifically ignore object dtype here if the inferred type is that it contains all strings, so users consistently get a columns Index object using `str` dtype

Reply:
Hmm that's tricky but I think going with the str data type as you suggested is better; I would expect that is a better UX in over 99% of instances

Reply:
OK, changed this to ensure we actually use `str` dtype columns Index object, even if the pandas metadata of the pyarrow table says that the original table was using object dtype. 

This ensures that all existing files will use (with pandas>= 3) the default str dtype for the columns, but that also has the trade-off that _if_ you explicitly want to use object dtype with strings, that this will no longer roundtrip in pandas->pyarrow/parquet->pandas)